dtfs-central-api/src/services/state-machines/utilisation-report/event-handlers/remove-fees-from-payment/remove-fees-from-payment.event-handler.ts

```diff
@@ -0,0 +1,58 @@
+import { EntityManager } from 'typeorm';
+import { DbRequestSource, FeeRecordEntity, UtilisationReportEntity } from '@ukef/dtfs2-common';
+import { BaseUtilisationReportEvent } from '../../event/base-utilisation-report.event';
+import { FeeRecordStateMachine } from '../../../fee-record/fee-record.state-machine';
+import { feeRecordsMatchAttachedPayments } from '../helpers';
+
+type RemoveFeesFromPaymentEventPayload = {
+  transactionEntityManager: EntityManager;
+  selectedFeeRecords: FeeRecordEntity[];
+  otherFeeRecords: FeeRecordEntity[];
+  requestSource: DbRequestSource;
+};
+
+export type UtilisationReportRemoveFeesFromPaymentEvent = BaseUtilisationReportEvent<'REMOVE_PAYMENT_FEES', RemoveFeesFromPaymentEventPayload>;
+
+const removeSelectedFeePayments = async (transactionEntityManager: EntityManager, feeRecords: FeeRecordEntity[], requestSource: DbRequestSource) => {
+  const feeRecordStateMachines = feeRecords.map((feeRecord) => FeeRecordStateMachine.forFeeRecord(feeRecord));
+  await Promise.all(
+    feeRecordStateMachines.map((stateMachine) =>
+      stateMachine.handleEvent({
+        type: 'REMOVE_FROM_PAYMENT',
+        payload: {
+          transactionEntityManager,
+          requestSource,
+        },
+      }),
+    ),
+  );
+};
+
+const updateOtherFeePaymentsInGroup = async (transactionEntityManager: EntityManager, feeRecords: FeeRecordEntity[], requestSource: DbRequestSource) => {
+  const feeRecordsAndPaymentsMatch = await feeRecordsMatchAttachedPayments(feeRecords, transactionEntityManager);
+
+  const feeRecordStateMachines = feeRecords.map((feeRecord) => FeeRecordStateMachine.forFeeRecord(feeRecord));
+  await Promise.all(
+    feeRecordStateMachines.map((stateMachine) =>
+      stateMachine.handleEvent({
+        type: 'OTHER_FEE_REMOVED_FROM_GROUP',
+        payload: {
+          transactionEntityManager,
+          feeRecordsAndPaymentsMatch,
+          requestSource,
+        },
+      }),
+    ),
+  );
+};
+
+export const handleUtilisationReportRemoveFeesFromPaymentEvent = async (
+  report: UtilisationReportEntity,
+  { transactionEntityManager, selectedFeeRecords, otherFeeRecords, requestSource }: RemoveFeesFromPaymentEventPayload,
+): Promise<UtilisationReportEntity> => {
+  await removeSelectedFeePayments(transactionEntityManager, selectedFeeRecords, requestSource);
+  await updateOtherFeePaymentsInGroup(transactionEntityManager, otherFeeRecords, requestSource);
+
+  report.updateLastUpdatedBy(requestSource);
+  return await transactionEntityManager.save(UtilisationReportEntity, report);
```

Comment:
This is interesting, first of all I'm not sure we are being consistent with this across different event handlers which is possibly an issue. IMO we aren't changing anything directly on the report entity so we shouldn't need to update the last updated by fields and save any changes on the report itself, but that is perhaps worth us validating with the full team.

@ChristianMcc @JoshBinns2000 do you have any thoughts on this?

Reply:
I've been working under the principal that, in general, any changes to anything linked to the utilisation report should result in updating the utilisation report audit fields as well. However, I don't have any particular reason for doing this and we don't do that for other linked entities, so I'm not sure whether or not we should be. I don't necessarily think there is a right answer, but it would be good for us to establish something more consistent than what we have at the moment

Reply:
Thinking of leaving this as it is as we're doing similar within the edit payment event handler.

I think it could be good if we could get more info on which type of events we should be updating the audit log for (if not all) if we're on the fence about adding this in some places.

Reply:
Yep let's leave this as is and I will follow up on slack for a more long term plan going forwards.